mqtt/mqtt-broker/src/auth.rs

```diff
@@ -0,0 +1,71 @@
+use derive_more::Display;
+use failure::Fail;
+
+use crate::Error;
+
+#[derive(Clone, Debug, Display, PartialEq)]
+pub enum AuthId {
+    #[display(fmt = "*")]
+    Any,
+
+    Value(Identity),
+}
+
+pub type Identity = String;
+
+#[derive(Clone, Debug)]
+pub struct Certificate(Vec<u8>);
```

Comment:
I'm assuming this will eventually be some more sophisticated type than a byte array? Do the TLS libraries give us a parsed certificate?

Reply:
TLS library provide it as a `Vec<u8>` we need to parse it with something else.

Reply:
bummer. :)

Reply:
Wait, so does the `Authenticator` trait need to be async then?

Reply:
Because it will call EH via some kind of rpc?

Reply:
yeah, that seems to imply the authentcator trait would need to be async then, if the rpc is going to happen in the `authenticate` call

Reply:
We also probably don't need to call into the EdgeHub for cert authentication, just username/password? Hmm, I guess there is the whole thumbprint cert auth and scopes though...